base/w3ctime.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "std/ctime.hpp"
+#include "std/string.hpp"
+
+namespace base
+{
+
+inline bool NotATime(time_t const t) noexcept { return t == -1; }
+
+// See http://www.w3.org/TR/NOTE-datetime to learn more
+// about a format.
+
+// Parse a string of a format YYYY-MM-DDThh:mmTZD
+// (eg 1997-07-16T19:20:30.45+01:00).
+// Return timestamp corresponding to w3ctime or -1 on failure.
+time_t ParseTime(std::string const & w3ctime) noexcept;
+
+// Convert timestamp to a string of a fromat YYYY-MM-DDThh:mmTZD
+std::string TimeToString(time_t const timestamp) noexcept;
```

Comment:
Returns an empty string when it's not possible to convert timestamp to a string?